index.md

```diff
@@ -0,0 +1,220 @@
+## Styleguide
+
+### Table of Contents
+
+- [1.0 General](#10-general)
+- [2.0 Angular / Coffeescript](#20-angular--coffeescript)
+  - [2.1 Basics](#21-basics)
+  - [2.2 Angular specifics](#22-angular-specifics)
+  - [2.3 Coffeescript specifics](#23-coffeescript-specifics)
+- [3.0 Django / Python](#30-django--python)
+  - [3.1 Basics](#31-basics)
+- [4.0 Styleguide](#40-styleguide)
+
+
+### 1.0 General
+
+- Avoid unnecessary else statements
+
+```python
+# bad
+if a_true_thing:
+    return True
+else:
+    return False
+
+# good
+if a_true_thing:
+    return True
+return False
+```
+
+- Avoid nested if statements
+
+```python
+# bad
+if a_true_thing:
+    if another_true_thing:
+        return 'foo'
+    else:
+        return 'bar'
+
+# good
+if a_true_thing and another_true_thing:
+    return 'foo'
+elif a_true_thing:
+    return 'bar'
+```
+
+### 2.0 Angular / Coffeescript
+
+#### 2.1 Basics
+
+- 2 space indent
+
+```coffeescript
+hello = ->
+  console.log('hi')
+  return
+hello()
+```
+
+- Single quotes for strings, unless using string interpolation, then double quotes is acceptable
+
+```coffeescript
+# bad
+aString = "Hello World"
+anotherString = '#{ aString }' # this won't work at all
+
+# good
+aString = 'Hello World'
+anotherString = "#{ aString }"
+```
+
+- Commas should have a space after them
+
+```coffeescript
+# bad
+list = [1,2,3,4]
+foo = (bar,baz) -> ...
+
+# good
+list = [1, 2, 3, 4]
+foo = (bar, baz) -> ...
+```
+
+- CamelCase when naming variables and functions
+
+```coffeescript
+# bad
+some_function = (foo, bar) -> ...
+some_var = 'Hello world'
+
+# good
+someFunction = (foo, bar) -> ...
+someVar = 'Hello world'
+```
+
+- No quotes on keys in key value pairs
+
+```coffeescript
+# bad
+obj = {'key': 'value'}
+
+# good
+obj = {key: 'value'}
```

Comment:
I actually vote for the bad case here. I don't like that we sometimes have quotes and other times don't. This feels like one of those things coffee gives us that isn't helpful and doesn't actually match python (and matching python is a lot of the value of coffee imo)

Reply:
so upon further discussion between joe and i, im going to move the dash exception to bad. you just shouldnt use dashes in key names cause its silly. if its coffeescript, it should be camelCase, and if its constructing something in coffee for backend consumption, it should be snake_case. dashes should almost never come into play

Reply:
Please add the coffee and backend consumption cases to `good`

Reply:
There are cases when it might be reasonable to have quoted keys - if we're testing something that generates objects with key/val pairs in which the key is based off of some arbitrary input, we might need to write something like:
```coffee
expectedOutput = {
 'expected foo': 'bar'
}
```

Reply:
You're right that there are exceptions, however they are few and far between. I don't know if it's really worth calling out, what do you think?

Reply:
That's probably fair. 👍